### ISSUE TITLE
GGMAX Bug - SpawnNewEntity causing MAX to crash?

### ISSUE THREAD

**Issue by AmenMoses** [TheGameCreators/GameGuruRepo#5738]
I have a script that uses the SpawnNewEntity command to make multiple copies of entities (the piano script much demoed on the discord) but when testing it MAX crashes regularly,

i.e. if I run test game for the first time it usually works fine (but not always) but subsequent test game runs tend to crash out with no error, just returns me to windows desktop.

I've checked the logs and nothing reported there either.

**Comment by Kasseyus** [comment]
@AmenMoses Hi there.  Can you please send me the FPM file in which you experience this issue - and the script you mentioned so I can attempt to recreate the conditions of the crash.

These can be sent to me at chris@thegamecreators.com

**Comment by AmenMoses** [comment]
@Kasseyus Don't have working email at present but here are the files you need to test the piano:

[entitybank.zip](https://github.com/user-attachments/files/17121187/entitybank.zip)
[scriptbank.zip](https://github.com/user-attachments/files/17121191/scriptbank.zip)
[audiobank.zip](https://github.com/user-attachments/files/17121192/audiobank.zip)

Just create an empty level, no biome or anything.
Drop in one of each entity, i.e. piano, piano_lid and four keys.
Then run test game.
Play with the piano to check it all works, then escape back to the editor and hit test game again.
For me running with the latest public version of MAX (which seems to be called 2024.06.18 ?) it usually will dump back out to windows, sometimes on the second test but definitely after a few more.

**Comment by korodai** [comment]
I noticed spawning new entities will sometimes crash MAX to desktop if DeleteNewEntity() is not called before exiting the script.

**Comment by Kasseyus** [comment]
@LeeBamberTGC 
Verified and consistent crashes on subsequent 'test games'
https://github.com/user-attachments/assets/e19c665f-ffd7-407a-b540-470043eb9928

Attached an  FPM file of a level with the  piano setup, but you'll need to download the files above.

[Mozart.zip](https://github.com/user-attachments/files/17356596/Mozart.zip)

Also.  This piano is totally awesome.

**Comment by AmenMoses** [comment]
> Also. This piano is totally awesome.

It's actually an oldie from the Classic days reworked in MAX to use the dynamic spawning for the entities rather than the user having to place 80 odd keys down in the level. The models were provided by GraPhiX and they and the script are freely available to anyone who wants them.

Glad you like it though.

**Comment by plemsoft** [comment]
This has been fixed and waiting for Lee to compile a new devtest , it was a physics memory problem missing allocation for spawned objects. Also added automatic cleanup if DeleteNewEntity() is not used.

**Comment by LeeBamberTGC** [comment]
@plemsoft New DEVTEST uploaded :)

**Comment by AmenMoses** [comment]
Is DEVTEST different from the Experimental build?

**Comment by Kasseyus** [comment]
@AmenMoses Yes.  This was introduced a few months back, as the EXP build was supposed to be where new fixes and features were deployed to, but it was by its very nature an unstable platform. Over time,  a lot of people were using the EXP build as their primary development platform and wanted stability AND the latest features.  Unfortunately, when a fix had unexpected consequences, it affected a lot of people and could potentially cause work to be lost.

The solution was to introduce the DevTest build; where fresh fixes get pushed out to a smaller group of users who understand that this platform is a testing area, and isn't designed for primary development.  Provided it don't rip holes in the fabric of space-time, or consume itself like a ravenous ouroboros, it then gets pushed to the EXP build for more people to enjoy newer features in a more stable way.

And of course, this is all then pushed onto the default build for everyone on a quarterly basis.  So far, it's proven quite effective. :)
